addon/mixins/ajax-request.js

```diff
@@ -225,15 +225,20 @@ export default Mixin.create({
    */
   request(url, options) {
     const hash = this.options(url, options);
-    return new Promise((resolve, reject) => {
-      this._makeRequest(hash)
-        .then(({ response }) => {
-          resolve(response);
-        })
-        .catch(({ response }) => {
-          reject(response);
-        });
+    const internalPromise = this._makeRequest(hash);
+
+    const ajaxPromise = new AJAXPromise((resolve, reject) => {
+      internalPromise.then(({ response }) => {
+        resolve(response);
+      })
+      .catch(({ response }) => {
+        reject(response);
+      });
     }, `ember-ajax: ${hash.type} ${hash.url} response`);
```

Comment:
Oh yeah, I don't doubt that the debugging help here is great.  However, we already add a label to the promise [here](https://github.com/alexlafroscia/ember-ajax/blob/a7094e5c32a68a8678c4e19d2abf807aea9c1711/addon/mixins/ajax-request.js#L335).  So, we create a second promise here just to add an additional label to the promise, when we could just as easily leave the additional promise and the label out.